tests/utils.py

```diff
@@ -123,27 +123,44 @@ def setUp(self):
         self.segn = torch.tensor(self.segn)
 
 
-def test_script_save(net, *inputs, eval_nets=True):
+def test_script_save(net, *inputs, eval_nets=True, device=None):
     """
     Test the ability to save `net` as a Torchscript object, reload it, and apply inference. The value `inputs` is
     forward-passed through the original and loaded copy of the network and their results returned. Both `net` and its
     reloaded copy are set to evaluation mode if `eval_nets` is True. The forward pass for both is done without
     gradient accumulation.
+
+    The test will be performed with CUDA if available, else CPU.
     """
+    if not device:
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    # Convert to device
+    inputs = [i.to(device) for i in inputs]
+    net.to(device)
 
     scripted = torch.jit.script(net)
     buffer = scripted.save_to_buffer()
-    reloaded_net = torch.jit.load(BytesIO(buffer))
+    reloaded_net = torch.jit.load(BytesIO(buffer)).to(device)
 
     if eval_nets:
         net.eval()
         reloaded_net.eval()
 
     with torch.no_grad():
+        set_determinism(seed=0)
         result1 = net(*inputs)
         result2 = reloaded_net(*inputs)
-
-    return result1, result2
+        set_determinism(seed=None)
+    # When using e.g., VAR, we will produce a tuple of outputs.
+    # Hence, convert all to tuples and then compare all elements.
+    if not isinstance(result1, tuple):
+        retult1 = (result1,)
+        retult2 = (result2,)
+
+    for i, j in zip(result1, result2):
+        if None not in (i, j):  # might be None
+            np.testing.assert_allclose(i.cpu(), j.cpu())
```

Comment:
when I modify this line to
`np.testing.assert_allclose(i.detach().cpu().numpy(), j.detach().cpu().numpy(), atol=1e-5)`
tests are passed for me

Reply:
something like this: https://github.com/YuanTingHsieh/MONAI/commit/81a337428878c6114d3698ad0e2d7907187d95e3